aicsimageio/writers/ome_tiff_writer.py

```diff
@@ -0,0 +1,641 @@
+                "Samples must be last dimension if present, and only S=3 or 4 is \
+                supported."
+            )
+        if dimension_order[-2:] != "YX" and dimension_order[-3:] != "YXS":
+            raise InvalidDimensionOrderingError(
+                f"Last characters of dimension_order {dimension_order} expected to \
+                be YX or YXS.  Please transpose your data."
+            )
+
+        # remember whether S was a dim or not, and remove it for now
+        if is_rgb:
+            ndims = ndims - 1
+            dimension_order = dimension_order[:-1]
+
+        # expand to 5D and add appropriate dimensions
+        if len(dimension_order) == 2:
+            dimension_order = "TCZ" + dimension_order
+
+        # expand to 5D and add appropriate dimensions
+        elif len(dimension_order) == 3:
+            # prepend either TC, TZ or CZ
+            if dimension_order[0] == "T":
+                dimension_order = "CZ" + dimension_order
+            elif dimension_order[0] == "C":
+                dimension_order = "TZ" + dimension_order
+            elif dimension_order[0] == "Z":
+                dimension_order = "TC" + dimension_order
+
+        # expand to 5D and add appropriate dimensions
+        elif len(dimension_order) == 4:
+            # prepend either T, C, or Z
+            first2 = dimension_order[:2]
+            if first2 == "TC" or first2 == "CT":
+                dimension_order = "Z" + dimension_order
+            elif first2 == "TZ" or first2 == "ZT":
+                dimension_order = "C" + dimension_order
+            elif first2 == "CZ" or first2 == "ZC":
+                dimension_order = "T" + dimension_order
+
+        return dimension_order, is_rgb
+
+    @staticmethod
+    def _size_of_ndarray(data: List[types.ArrayLike]) -> int:
+        """
+        Calculate the size of data to determine if we require bigtiff
+
+        Parameters
+        ----------
+        data: list of data arrays, one per image to be saved to tiff
+
+        Returns
+        -------
+        the total size of data in bytes
+        """
+        size = 0
+        for i in range(len(data)):
+            size += data[i].size * data[i].itemsize
+        return size
+
+    @staticmethod
+    def _extend_data_shape(shape: Tuple[int, ...], num_dims: int) -> Tuple[int, ...]:
+        # extend data shape to be same len as dimension_order
+        if len(shape) < num_dims:
+            shape = tuple([1] * (num_dims - len(shape))) + shape
+        return shape
+
+    @staticmethod
+    def _build_ome_image(
+        image_index: int = 0,
+        tiff_plane_offset: int = 0,
+        data_shape: Tuple[int, ...] = (1, 1, 1, 1, 1),
+        data_dtype: np.dtype = np.dtype(np.uint8),
+        is_rgb: bool = False,
+        dimension_order: str = DEFAULT_DIMENSION_ORDER,
+        image_name: Optional[str] = "I0",
+        pixels_physical_size: Tuple[float, float, float] = (1.0, 1.0, 1.0),
```

Comment:
Same comment from above, can we change this parameter to `physical_pixel_sizes` for consistency?